src/encoded/audit/analysis.py

```diff
@@ -291,24 +736,1430 @@ def check_analysis_dnase_seq_standards(
+    target,
+    read_depth,
+    link_to_standards
+):
+    if target and 'name' in target:
+        target_name = target['name']
+    elif control_type:
+        target_name = control_type
+    else:
+        return
+
+    modERN_cutoff = pipelines_with_read_depth[
+        'Transcription factor ChIP-seq pipeline (modERN)']
+    if not read_depth:
+        detail = (
+            f"ENCODE processed {file_to_check['output_type']} file "
+            f"{audit_link(path_to_text(file_to_check['@id']), file_to_check['@id'])} "
+            f"has no read depth information."
+        )
+        yield AuditFailure('missing read depth', detail, level='INTERNAL_ACTION')
+        return
+
+    if read_depth < modERN_cutoff:
+        first_half_of_detail = (
+            f"modERN processed alignment file "
+            f"{audit_link(path_to_text(file_to_check['@id']), file_to_check['@id'])} "
+            f"has {read_depth} usable fragments. "
+        )
+        if control_type == 'input library':
+            detail = (
+                f"{first_half_of_detail} It cannot be used as a control "
+                f"in experiments studying transcription factors, which "
+                f"require {modERN_cutoff} usable fragments, according to "
+                f"the standards defined by the modERN project."
+            )
+        else:
+            detail = (
+                f"{first_half_of_detail} Replicates for ChIP-seq assays and "
+                f"target {target_name} investigated as transcription factor "
+                f"require {modERN_cutoff} usable fragments, according to "
+                f"the standards defined by the modERN project."
+            )
+        yield AuditFailure('insufficient read depth', detail, level='NOT_COMPLIANT')
+
+    return
+
+
+'''
+RNA-seq and transcription assay audits
+'''
+
+
+def audit_missing_read_depth(file):
+    detail = (
+        f"Processed {file['output_type']} file "
+        f"{audit_link(path_to_text(file['@id']), file['@id'])} "
+        f"has no read depth information."
+    )
+    yield AuditFailure('missing read depth', detail, level='INTERNAL_ACTION')
+    return
+
+
+def audit_missing_star_quality_metric(value, alignment_files, pipeline):
+    file_ids = [f['@id'] for f in alignment_files]
+    file_links = [audit_link(path_to_text(file), file) for file in file_ids]
+    detail = (
+        f"Alignment file(s) {', '.join(file_links)} in "
+        f"{audit_link(path_to_text(value['@id']), value['@id'])} processed by "
+        f"{audit_link(path_to_text(value['pipelines'][0]['title']), value['pipelines'][0]['@id'])} "
+        f"have no read depth containing quality metric associated with it."
+    )
+    yield AuditFailure('missing read depth', detail, level='INTERNAL_ACTION')
+
+
+def check_analysis_bulk_rna_standards(
+    value,
+    files_structure,
+    expected_pipeline_titles,
+    link_to_standards,
+):
+    if value['assembly'] not in ['GRCh38', 'mm10']:
+        return
```

Comment:
Is this check universal? ChIP-seq really doesn't check this?

Reply:
Not universal. I tried to stay consistent with whatever was in the original Experiment audit, and some exclude hg19 and others do not.

Reply:
Thank you for clarifying. Good to stay this way then.